Casks/n/nvidia-nsight-systems.rb

```diff
@@ -0,0 +1,24 @@
+cask "nvidia-nsight-systems" do
+  version "2024.7.1.84-3512561"
+  sha256 "9bc4027383649845f8f4ffe5bc9c30aaffbb83167f21c92504ffb8532ea5e018"
+
+  url "https://developer.nvidia.com/downloads/assets/tools/secure/nsight-systems/#{version.major}_#{version.minor}/NsightSystems-macos-public-#{version}.dmg"
+  name "NVIDIA Nsight Systems"
+  desc "System-wide performance analysis tool"
+  homepage "https://developer.nvidia.com/nsight-systems"
+
+  livecheck do
+    url "https://developer.nvidia.com/tools-downloads.json" # https://developer.nvidia.com/nvidia-cuda-toolkit-developer-tools-mac-hosts
+    regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]\d+)+)\.dmg/i)
```

Comment:
```suggestion
    regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]\d+)+)\.dmg/i)
    strategy :json do |json, regex|
      json["downloads"]&.map do |download|
        next unless download["development_platform"]&.include?("osx")

        download["files"]&.map do |file|
          match = file["url"]&.match(regex)
          next if match.blank?

          match[1]
        end
      end.flatten
    end
```

We can be more specific by using the `Json` strategy, which is suitable for handling JSON responses.

Reply:
Indeed. I've tried using json strategy but didn't figure it out, for I'm not familiar with ruby. Thanks!